### PR TITLE
use hostname from auth_url in 3rd party cookie message

### LINKED ISSUE #4022: 3rd party cookies disabled message doesn't show delegated domain
mozilla.com delegates to login.mozilla.org, but if a user doesn't accept 3rd party cookies, they are told to whitelist mozilla.com.

### PR DESCRIPTION
**Pull request by seanmonstar**
fixes #4022

### CODE REVIEW

File: resources/static/dialog/js/misc/state.js [hunk 1]
```diff
@@ -312,9 +312,14 @@ BrowserID.State = (function() {
         add: !!self.addPrimaryUser,
         email: self.email,
         rpInfo: self.rpInfo,
-        idpName: info.idpName || URLParse(info.auth_url).host
+        idpName: info.idpName,
+        idpHost: URLParse(info.auth_url).host
       });
 
+      if (info.idpName) {
```
Comment (fmarier) [inline resources/static/dialog/js/misc/state.js]: Are you sure you don't mean the opposite here? `if (!info.idpName) {`

Comment (seanmonstar) [inline resources/static/dialog/js/misc/state.js]: yea, you're right. woops.

**Comment by seanmonstar** [comment]
ok, reversed that if clause.

**Comment by fmarier** [comment]
Looks good!

**Pull request merged**